crates/db_schema/src/impls/community.rs

```diff
@@ -194,6 +194,38 @@ impl DeleteableOrRemoveable for Community {
   }
 }
 
+pub enum CollectionType {
+  Moderators,
+  Featured,
+}
+
+impl Community {
+  /// Get the community which has a given moderators or featured url, also return the collection type
+  pub async fn get_by_collection_url(
```

Comment:
This should really be two functions. Then instead of creating an enum and using it as a parameter, you can just do:

`fn get_by_moderators_url(` , and `fn get_by_featured_url(`.

Reply:
When this is called we dont know which type of url it is, we need to determine that from the db. Thats why this signature makes the most sense, otherwise every callsite would need to copy the same logic.

Reply:
What if a community has both a moderators and a featured_url ? In this case your calls would always pull the moderators one. 

Isn't there any way to know / differentiate that type before you get here?

Reply:
Thats not how this works. The code is used when we receive an `Add`  or `Remove` activity, which modifies a collection. Normally we distinguish activities based on the Rust typesystem, for example `Like` and `Dislike` have a different type for the `kind` field so they can be parsed by serde into different structs. This is not possible here because "add moderator" and "add favourite post" both look identical in terms of Rust types. So we need to make this database lookup to find the matching entry, and determine which community it belongs to. This is necessary because other federated platforms can use different URL formats from Lemmy, so we cant do something like parse the URL itself.